src/development/platform-integration/android/restore-state-android.md

```diff
@@ -0,0 +1,212 @@
+
+The `RestorationManager` manages the restoration
+data, which is serialized to a tree of [`RestorationBucket`][],
+that are synchronized with the Flutter engine.
+A `RestorationBucket` contains a key/value pair
+of state information that you want to save. The key is a
+[`restorationId`][] that is unique to that `RestorationScope`.
+
+Flutter automatically creates a
+[`RestorationScope`][] for each route in your
+app to track the `RestorationBucket`s in that scope.
+In general, you shouldn't need to worry about
+this object _unless_ the scope has two (or more) buckets with
+the same `restorationId`. If that occurs,
+create a new `RestorationScope` to ensure
+that each ID is unique to its scope.
+Each scope is assigned a unique [`restorationScopeId`][].
+
+You must decide what state you want to save and restore,
+but you can only save specific types to a `RestorationBucket`,
+namely: `null`, `bool`, `int`, `double`, `String`, `Uint8List`
+(and other typed data), `List`, `Map`, and child `RestorationBucket`s.
+
+{{site.alert.note}}
+  The code in this page was taken from [VeggieSeasons][].
+  VeggieSeasons is a sample app written for iOS that uses Cupertino
+  widgets. An iOS app requires [a bit of extra setup][] in Xcode, but
+  the restoration classes otherwise work the same on both iOS and Android.
+{{site.alert.end}}
+
+
+[a bit of extra setup]: {{site.api}}/flutter/services/RestorationManager-class.html#state-restoration-on-ios
+[`restorationId`]: {{site.api}}/flutter/widgets/RestorationScope/restorationId.html
+[`restorationScopeId`]: {{site.api}}/flutter/widgets/RestorationScope/restorationScopeId.html
+[`RestorationScope`]: {{site.api}}/flutter/widgets/RestorationScope-class.html
+[VeggieSeasons]: {{site.github}}/flutter/samples/tree/master/veggieseasons
+
+## Step 1: Enabling state restoration
+
+Most every Flutter app contains a `main` method that calls the
+[`runApp`][] function. You can use `runApp`
+to set up the framework and the bindings that bind
+the Dart code to the native Flutter engine.
+Set up the `RestorationManager` in the `runApp` function.
+For example:
+
+```dart
+void main() {
+  /...
+
+  runApp(
+    const RootRestorationScope(
+      restorationId: 'root',
+      child: VeggieApp(),
+    ),
+  );
+}
+```
+
+QUESTION for Michael: I thought that Flutter automatically created
+  the Scope for every route. But this app manually
+  creats a Scope for the page. ?? (xxx)
+
+[`runApp`]: {{site.api}}/flutter/widgets/runApp.html
+
+## Step 2: Add RestorationMixin to the State class
+
+To implement restorable state,
+add the [`RestorationMixin`][] to the app's `State` class.
+
+```dart
+class _VeggieAppState extends State<VeggieApp> with RestorationMixin
+```
+
+Once you've added the restore state mixin,
+your IDE asks you to implement an abstract method
+called [`restoreState`][]. The `restoreState` method calls
+[`registerForRestoration`][] to register that you
+intend to save state. For example:
```

Comment:
What I feel is missing here that you can only restore state with the RestorationMixin that is stored in RestorableProperty (and their subclasses). To restore the state in those properties, you have to register them in restoreState.